### PR TITLE
reicast: remove && flycast: init at 1.2

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Description of changes
I noticed reicast's upstream repository has been archived, so I've removed it and added a maintained fork as [upstream suggests](https://github.com/skmp/reicast-emulator/tree/95b673698863b6a8a4e1852ec018855cca43f9c7#project-eol).

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by asymmetric — CHANGES_REQUESTED** [review]
This PR has unfortunately accumulated merge conflicts.

**Comment by IvarWithoutBones** [comment]
> This PR has unfortunately accumulated merge conflicts.

Resolved them :+1:

File: pkgs/applications/emulators/flycast/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    SDL2
+    libzip
+    miniupnpc
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/flyinghead/flycast";
+    changelog = "https://github.com/flyinghead/flycast/releases/tag/v${version}";
+    description = "A multi-platform Sega Dreamcast, Naomi and Atomiswave emulator";
+    mainProgram = "flycast";
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/flycast/default.nix:48]:
```suggestion
```
that is the default

File: pkgs/applications/emulators/flycast/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+stdenv.mkDerivation rec {
+  pname = "flycast";
+  version = "1.2";
+
+  src = fetchFromGitHub {
+    owner = "flyinghead";
+    repo = "flycast";
+    rev = "v${version}";
+    sha256 = "sha256-MzHAGK++oukIs84OR/l6gBwCJssdi8Iyte5Rtro2+Q0=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/flycast/default.nix:25]: Did you make sure we use our SDL?

Comment (IvarWithoutBones) [reply pkgs/applications/emulators/flycast/default.nix:25]: Yes:
```
flycast> -- Found SDL2: /nix/store/xqikpb1nlipsj6jqvn03z1z969jb404v-SDL2-2.0.14/lib/libSDL2.so
```

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**